src/include/compat.h

```diff
@@ -110,4 +112,18 @@
 #define MSG_DONTWAIT MSG_NONBLOCK
 #endif
 
+#if defined(HAVE_PTHREAD_SETNAME_NP)
+#define PTHREAD_SETNAME_NP(thread, name) pthread_setname_np(thread, name)
+#elif defined(HAVE_PTHREAD_SET_NAME_NP)
+#define PTHREAD_SETNAME_NP(thread, name) pthread_set_name_np(thread, name)
+#else
+#define PTHREAD_SETNAME_NP(thread, name)
+#endif
+
+#if defined(HAVE_PTHREAD_GETNAME_NP)
+#define PTHREAD_GETNAME_NP(thread, name, len) pthread_getname_np(thread, name, len)
+#else
+#define PTHREAD_GETNAME_NP(thread, name, len)
```

Comment:
@dx9 
You might want to have the PTHREAD_GETNAME_NP macro return 0 and have it the first byte of name to '\0'
You can then remove the conditional around the usuage of this macro